sources/mongodb/helpers.py

```diff
@@ -37,40 +37,62 @@ def __init__(
         self.client = client
         self.collection = collection
         self.incremental = incremental
+
         if incremental:
             self.cursor_field = incremental.cursor_path
-            self.last_value = incremental.last_value
         else:
-            self.cursor_column = None
-            self.last_value = None
+            self.cursor_field = None
 
     @property
     def _filter_op(self) -> Dict[str, Any]:
-        if not self.incremental or not self.last_value:
+        """Build a filter operator.
+
+        Describes what logical statement to apply to which field.
+
+        Returns:
+            Dict[str, Any]: The filter operator.
+        """
+        if not self.incremental:
             return {}
-        if self.incremental.last_value_func is max:
-            return {self.cursor_field: {"$gte": self.last_value}}
-        elif self.incremental.last_value_func is min:
-            return {self.cursor_field: {"$lt": self.last_value}}
-        return {}
 
-    def load_documents(self) -> Iterator[TDataItem]:
-        cursor = self.collection.find(self._filter_op)
-        while docs_slice := list(islice(cursor, CHUNK_SIZE)):
-            yield map_nested_in_place(convert_mongo_objs, docs_slice)
+        if self.incremental.start_value:
+            return {self.cursor_field: {"$gte": self.incremental.start_value}}
 
+        if self.incremental.end_value:
+            return {self.cursor_field: {"$lte": self.incremental.end_value}}
+
+        return {}
 
-class CollectionLoaderParallell(CollectionLoader):
     @property
     def _sort_op(self) -> List[Optional[Tuple[str, int]]]:
-        if not self.incremental or not self.last_value:
+        """Build a sorting operator.
+
+        Describes what sorting to apply to which field.
+
+        Returns:
+            List[Optional[Tuple[str, int]]]: The sorting operator.
+        """
+        if not (self.incremental and self.incremental.row_order):
             return []
-        if self.incremental.last_value_func is max:
+
+        if self.incremental.row_order == "asc":
```

Comment:
same here, min and max should reverse the order. `max` is the normal order btw. (ascending)

Reply:
we also need test for `min` and `max`. also check if incremental loading works